src/lazy/binary/raw/value.rs

```diff
@@ -173,39 +343,27 @@ impl<'top> LazyRawBinaryValue_1_0<'top> {
     }
 
     /// Returns the encoded byte slice representing this value's data.
-    fn value_body(&self) -> IonResult<&'top [u8]> {
+    pub fn value_body(&self) -> &'top [u8] {
         let value_total_length = self.encoded_value.total_length();
-        if self.input.len() < value_total_length {
-            eprintln!("[value_body] Incomplete {:?}", self);
-            return IonResult::incomplete(
-                "only part of the requested value is available in the buffer",
-                self.input.offset(),
-            );
-        }
-        let value_body_length = self.encoded_value.value_length();
+        debug_assert!(self.input.len() >= value_total_length);
+        let value_body_length = self.encoded_value.value_body_length();
         let value_offset = value_total_length - value_body_length;
-        Ok(self.input.bytes_range(value_offset, value_body_length))
+        self.input.bytes_range(value_offset, value_body_length)
```

Comment:
🗺️ An earlier version of the lazy reader API did not have the invariant that the top level value was fully in the buffer, so the `value_body` method would confirm it on each call. Now it's always true -- it's checked when the lazy value is constructed.